crates/solver/src/solver/http_solver.rs

```diff
@@ -513,10 +532,31 @@ impl Solver for HttpSolver {
             serde_json::to_string_pretty(&settled).unwrap()
         );
 
+        // verify solution is not empty
         if settled.orders.is_empty() {
             return Ok(vec![]);
         }
 
+        // verify internal custom interactions return only bufferable tokens to settlement contract
+        let non_bufferable_tokens = non_bufferable_tokens_used(
+            &settled.interaction_data,
+            &self.market_makable_token_list.addresses(),
+        );
+        if !non_bufferable_tokens.is_empty() {
+            tracing::debug!(
```

Comment:
Maybe we should `WARN` here as this is a violation of rules that should never happen?

Reply:
Sure.